sdk/communication/communication-phone-numbers/test/public/siprouting/utils/recordedClient.ts

```diff
@@ -117,3 +119,18 @@ export async function createRecordedClientWithToken(
   // casting is a workaround to enable min-max testing
   return { client, recorder };
 }
+
+export async function clearSipConfiguration(): Promise<void> {
+  const client = new SipRoutingClient(env.COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING ?? "");
+  await client.setRoutes([]);
+  await client.setTrunks([]);
+}
+
+export function getUniqueFqdn(order: string): string {
+  const length = 12;
+  let random = 0;
+  do {
+    random = Math.floor(Math.random() * 10 ** length);
+  } while (random < 10 ** (length - 1));
+  return `${order}.${random}.com`;
```

Comment:
You should consider using the [`recorder.variable` API](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/test-utils/recorder#recordervariable) so that you don't have to add a sanitizer for the FQDN. This might look something like:

```ts
let fqdnNumber = 1;
export function getUniqueFqdn(order: string) {
  // ... impl
  return recorder.variable(`fqdn-${fqdnNumber++}`, /* randomly generated FQDN value */);
}
```

You might also be able to remove the `compareBodies: false` option passed into the matcher after making this change.

Reply:
Used a recorder.variable and removed the FQDN sanitizer.

let fqdnNumber = 1;
export function getUniqueFqdn(recorder: Recorder): string {
  const uniqueDomain = uuid().replace(/-/g, "");
  return recorder.variable(`fqdn-${fqdnNumber++}`, `test.${uniqueDomain}.com`);
}
export function resetUniqueFqdns(): void {
  fqdnNumber = 1;
}

I was able to remove the `compareBodies: false`. Thank you.

I needed to add the `resetUniqueFqdns` method to be run after tests to ensure there will be variables in recordings always starting with '1' prefix. Without it a recording could contain variables like `fqdn-58`, `fqdn-59`, `fqdn-60`. Which caused problems with unexpected variables in browser tests where the tests classes run in slightly different order.